### DISCUSSION TITLE
a mac can not connect to the server,need help!

### DISCUSSION THREAD

**Discussion by fflrichard** [strongswan/strongswan#1551] [Q&A]
I have a mac that can not connect to the server, but other Macs can connect to it. here is the log, can anyone help?

root@ubuntu:~# swanctl --log
09[NET] received packet: from xx.xx.xx.xx[900] to 192.168.1.16[500] (432 bytes)
09[ENC] parsed IKE_SA_INIT request 0 [ SA KE No N(REDIR_SUP) N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) ]
09[IKE] xx.xx.xx.xx is initiating an IKE_SA
09[CFG] selected proposal: IKE:AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048
09[IKE] local host is behind NAT, sending keep alives
09[IKE] remote host is behind NAT
09[ENC] generating IKE_SA_INIT response 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(CHDLESS_SUP) N(MULT_AUTH) ]
09[NET] sending packet: from 192.168.1.16[500] to xxx[900] (456 bytes)
16[NET] received packet: from xx.xx.xx.xx[58605] to 192.168.1.16[4500] (368 bytes)
16[ENC] unknown attribute type INTERNAL_DNS_DOMAIN
16[ENC] parsed IKE_AUTH request 1 [ IDi N(INIT_CONTACT) IDr CPRQ(ADDR MASK DHCP DNS ADDR6 DHCP6 DNS6 DOMAIN) N(ESP_TFC_PAD_N) N(NON_FIRST_FRAG) SA TSi TSr N(MOBIKE_SUP) N(EAP_ONLY) ]
16[CFG] looking for peer configs matching 192.168.1.16[my.vpn.com]...xx.xx.xx.xx[gutao]
16[CFG] selected peer config 'conn'
16[IKE] initiating EAP_IDENTITY method (id 0x00)
16[IKE] received ESP_TFC_PADDING_NOT_SUPPORTED, not using ESPv3 TFC padding
16[IKE] peer supports MOBIKE
16[IKE] authentication of 'my.vpn.com' (myself) with RSA signature successful
16[IKE] sending end entity cert "CN=my.vpn.com"
16[ENC] splitting IKE message (3776 bytes) into 4 fragments
16[ENC] generating IKE_AUTH response 1 [ IDr CERT CERT AUTH EAP/REQ/ID ]
16[ENC] generating IKE_AUTH response 1 [ EF(1/4) ]
16[IKE] sending issuer cert "C=AT, O=ZeroSSL, CN=ZeroSSL RSA Domain Secure Site CA"
16[ENC] generating IKE_AUTH response 1 [ EF(2/4) ]
16[ENC] generating IKE_AUTH response 1 [ EF(3/4) ]
16[ENC] generating IKE_AUTH response 1 [ EF(4/4) ]
16[NET] sending packet: from 192.168.1.16[4500] to xx.xx.xx.xx[58605] (1236 bytes)
16[NET] sending packet: from 192.168.1.16[4500] to xx.xx.xx.xx[58605] (1236 bytes)
16[NET] sending packet: from 192.168.1.16[4500] to xx.xx.xx.xx[58605] (292 bytes)
16[NET] sending packet: from 192.168.1.16[4500] to xx.xx.xx.xx[58605] (1236 bytes)
08[IKE] sending keep alive to xx.xx.xx.xx[58605]
07[JOB] deleting half open IKE_SA with xx.xx.xx.xx after timeout

**Comment by misch42** [comment]
On 23.02.23 08:49, Ricahrd Fan wrote:
>
> I have a mac that can not connect to the server, but other Macs can 
> connect to it. here is the log, can anyone help?
>
> ***@***.***:~# swanctl --log
> 09[NET] received packet: from xx.xx.xx.xx[900] to 192.168.1.16[500] 
> (432 bytes)
> 09[ENC] parsed IKE_SA_INIT request 0 [ SA KE No N(REDIR_SUP) 
> N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) ]
> 09[IKE] xx.xx.xx.xx is initiating an IKE_SA
> 09[CFG] selected proposal: 
> IKE:AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048
> 09[IKE] local host is behind NAT, sending keep alives
> 09[IKE] remote host is behind NAT
> 09[ENC] generating IKE_SA_INIT response 0 [ SA KE No N(NATD_S_IP) 
> N(NATD_D_IP) N(FRAG_SUP) N(CHDLESS_SUP) N(MULT_AUTH) ]
> 09[NET] sending packet: from 192.168.1.16[500] to xxx[900] (456 bytes)
> 16[NET] received packet: from xx.xx.xx.xx[58605] to 192.168.1.16[4500] 
> (368 bytes)
> 16[ENC] unknown attribute type INTERNAL_DNS_DOMAIN
> 16[ENC] parsed IKE_AUTH request 1 [ IDi N(INIT_CONTACT) IDr CPRQ(ADDR 
> MASK DHCP DNS ADDR6 DHCP6 DNS6 DOMAIN) N(ESP_TFC_PAD_N) 
> N(NON_FIRST_FRAG) SA TSi TSr N(MOBIKE_SUP) N(EAP_ONLY) ]
> 16[CFG] looking for peer configs matching 
> 192.168.1.16[my.vpn.com]...xx.xx.xx.xx[gutao]
> 16[CFG] selected peer config 'conn'
> 16[IKE] initiating EAP_IDENTITY method (id 0x00)
> 16[IKE] received ESP_TFC_PADDING_NOT_SUPPORTED, not using ESPv3 TFC 
> padding
> 16[IKE] peer supports MOBIKE
> 16[IKE] authentication of 'my.vpn.com' (myself) with RSA signature 
> successful
> 16[IKE] sending end entity cert "CN=my.vpn.com"
> 16[ENC] splitting IKE message (3776 bytes) into 4 fragments
> 16[ENC] generating IKE_AUTH response 1 [ IDr CERT CERT AUTH EAP/REQ/ID ]
> 16[ENC] generating IKE_AUTH response 1 [ EF(1/4) ]
> 16[IKE] sending issuer cert "C=AT, O=ZeroSSL, CN=ZeroSSL RSA Domain 
> Secure Site CA"
> 16[ENC] generating IKE_AUTH response 1 [ EF(2/4) ]
> 16[ENC] generating IKE_AUTH response 1 [ EF(3/4) ]
> 16[ENC] generating IKE_AUTH response 1 [ EF(4/4) ]
> 16[NET] sending packet: from 192.168.1.16[4500] to xx.xx.xx.xx[58605] 
> (1236 bytes)
> 16[NET] sending packet: from 192.168.1.16[4500] to xx.xx.xx.xx[58605] 
> (1236 bytes)
> 16[NET] sending packet: from 192.168.1.16[4500] to xx.xx.xx.xx[58605] 
> (292 bytes)
> 16[NET] sending packet: from 192.168.1.16[4500] to xx.xx.xx.xx[58605] 
> (1236 bytes)
> 08[IKE] sending keep alive to xx.xx.xx.xx[58605]
> 07[JOB] deleting half open IKE_SA with xx.xx.xx.xx after timeout
>

The log states that your VPN server sends out the AUTH response packet 
to the client, but does not receive any further packet. Please check 
that the client receives this answer packet and sends the next packet.


Then search why one packet goes missing. This might be a bad firewall, 
bad NAT, bad confg, bad fragmentation handling, or something else.


But to be sure, find out what packet (server - client) or (client -> 
server) is dropped and then find out why.
